### PR TITLE
copr: Add more metrics for waiting stages

### PR DESCRIPTION
**Pull request by breeswish**
Signed-off-by: Breezewish <breezewish@pingcap.com>

### What problem does this PR solve?

Problem Summary: 

Now we are not able to know the details of waiting stages.

### What is changed and how it works?

What's Changed:

This PR adds more metrics for coprocessor waiting stages.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Manual test (WIP)

### Release note

### CODE REVIEW

File: src/coprocessor/tracker.rs [hunk 1]
```diff
@@ -38,12 +44,17 @@ enum TrackerState {
 #[derive(Debug)]
 pub struct Tracker {
     request_begin_at: Instant,
+    scheduled_at: Instant,
```
Comment (sticnarf) [inline src/coprocessor/tracker.rs:47]: How about storing these `Instant`s in `TrackerState`?

Comment (breeswish) [reply src/coprocessor/tracker.rs:47]: Good idea!

File: src/coprocessor/tracker.rs [hunk 2]
```diff
@@ -43,7 +49,10 @@ pub struct Tracker {
 
     // Intermediate results
     current_stage: TrackerState,
-    wait_time: Duration,
+    wait_time: Duration,          // Total wait time
+    schedule_wait_time: Duration, // Wait time spend on waiting for scheduling
+    snapshot_wait_time: Duration, // Wait time spend on waiting for a snapshot
+    handler_build_time: Duration, // Time spend on building the handler, not included in total wait time
```
Comment (sticnarf) [inline src/coprocessor/tracker.rs:55]: `spend` -> `spent`

**Pull request merged**